### PR TITLE
0.14.0 release

### PR DESCRIPTION
**Pull request by diego-plan9**
### Summary

Bump the version to `0.14.0`, due to the `0.4.0` release of the IBMQProvider.

### CODE REVIEW

**Comment by nonhermitian** [comment]
I think @chriseclectic wants to get an Aer bump out with this.  And we (@mtreinish) should get a Terra release out as well so that we can get the docs pushed up along with other fixes.

**Comment by mtreinish** [comment]
While I'd like to get docs updates merged sooner than later, a new terra release is at least a few weeks away, there are still things that have merged but haven't finished yet (also a terra release auto opens a PR doing this so it wouldn't be handled here).

**Comment by diego-plan9** [comment]
Bumped the provider to `0.4.1` - and removed the `on hold` label, which was added as an extra-measure for preventing accidental merging while ironing out the last bits. I think we are good now on the provider front!

**Comment by nonhermitian** [comment]
https://github.com/Qiskit/qiskit-ibmq-provider/issues/450

**Comment by nonhermitian** [comment]
https://github.com/Qiskit/qiskit-ibmq-provider/issues/451

File: docs/release_notes.rst [hunk 1]
```diff
@@ -21,6 +21,148 @@ This table tracks the meta-package versions and the version of each Qiskit eleme
...
+
+Terra 0.10.0
+============
+
+No Change
+
+Aer 0.3
+=======
+
+No Change
```
Comment (chriseclectic) [inline docs/release_notes.rst:564]: Should the changelog for Aer 0.3.3 be here?

Comment (mtreinish) [reply docs/release_notes.rst:564]: We normally just leave it at the minor version in the release notes, if there's no update. But that's mostly do to not having anything written normally for patch version release. The specific version included in the release will be in the version history table that gets generated at the top of this file. If there are release notes for Aer 0.3.3 we can easily add those here

**Comment by diego-plan9** [comment]
On the provider side, a new version is likely to be released on Monday - @jyu00 and others, please just commit directly to the PR when the time comes or if further changes are needed :+1:

**Comment by jyu00** [comment]
Provider `0.4.2` is out. @mtreinish @chriseclectic can you update this PR (I don't have permission) and maybe we can get the qiskit release out soon? Thanks.

**Comment by jyu00** [comment]
Unfortunately we're going to need another provider release. The API is rolling out error message/code soon, which is being picked up by the job schema as a `SimpleNamespace` type, but the current provider code assumes it's a `dict` type. I'll update when it's fixed.

File: docs/release_notes.rst [hunk 2]
```diff
@@ -21,6 +21,148 @@ This table tracks the meta-package versions and the version of each Qiskit eleme
...
+
+Ignis 0.2
+=========
+
+No Change
+
+Aqua 0.5
```
Comment (jyu00) [inline docs/release_notes.rst]: This should be Aqua 0.6

**Comment by jyu00** [comment]
ok provider `0.4.3` is out, hopefully this is it. @mtreinish Please update this PR again with the new release. Thank you!!

File: docs/release_notes.rst (hunk 2 above)
Comment (diego-plan9) [reply docs/release_notes.rst]: Good catch! The block was mostly a copy-paste from the `0.13` section, which was mentioning Aqua 0.5. Tweaked both in d5cad61afa6e4530ced596fa642c704703a791c6

**Comment by diego-plan9** [comment]
Bumped to 0.4.3!

**Review by mtreinish — DISMISSED** [review]

**Comment by chriseclectic** [comment]
Can we do a separate release for qiskit aer? This has been ready to go for 3 weeks now.

**Comment by jyu00** [comment]
provider 0.4.4 is coming shortly

**Comment by jyu00** [comment]
provider 0.4.4 is out. Thanks @mtreinish!

**Pull request merged**